frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/factorial-experiment-design.component.ts

```diff
@@ -473,22 +471,30 @@ export class FactorialExperimentDesignComponent implements OnInit, OnChanges, On
     }
   }
 
-  validateFactorNames(factors: ExperimentFactorFormData[]) {
+  isFactorTableRowValid(factorRowIndex: number): boolean {
+    const factors = this.factorialExperimentDesignForm.value.factors;
```

Comment:
there's a convenience getter function that already holds a reference to the factors that you could use. i.e. `this.factors` should be equivalent to `const factors = this.factorialExperimentDesignForm.value.factors;`